streams-tests/jvm/src/test/scala/zio/stream/ChunkSpec.scala

```diff
@@ -276,6 +276,48 @@ object ChunkSpec extends ZIOBaseSpec {
       assert(Chunk(1, 2, 3).zipAllWith(Chunk(3, 2, 1))(_ => 0, _ => 0)(_ + _))(equalTo(Chunk(4, 4, 4))) &&
       assert(Chunk(1, 2, 3).zipAllWith(Chunk(3, 2))(_ => 0, _ => 0)(_ + _))(equalTo(Chunk(4, 4, 0))) &&
       assert(Chunk(1, 2).zipAllWith(Chunk(3, 2, 1))(_ => 0, _ => 0)(_ + _))(equalTo(Chunk(4, 4, 0)))
+    },
+    testM("flatMap Non Empty") {
+      val fn = Gen.function[Random with Sized, Int, Chunk[Int]](smallChunks(intGen))
+      check(smallChunks(intGen), fn) { (c_, f_) =>
+        val c: NonEmptyChunk[Int]        = c_ + 0
+        val f: Int => NonEmptyChunk[Int] = f_.andThen(_ + 0)
+
+        val in: NonEmptyChunk[Int] = c.flatMap(f)
+        val expected: Seq[Int]     = c.toSeq.flatMap(f.andThen(_.toSeq))
+
+        assert(in.toSeq)(equalTo(expected))
+      }
+    },
+    testM("nonEmptyChunk subtype") {
```

Comment:
If we just want to check that the subtype is preserved we probably don't need to generate a bunch of random data. We could do something like:
```scala
def assertNonEmpty[A](value: => NonEmptyChunk[A]): TestResult = {
  lazy val _ = value
  assertCompletes
}

lazy val nonEmptyChunk: NonEmptyChunk[A] = ???

suite("NonEmptyChunk subtype preservation)(
  zio.test.test("map")(nonEmptyChunk.map(identity))
  zio.test.test("flatMap")(nonEmptyChunk.flatMap(ZIO.succeed(_)))
  // All other combinators defined on subtype and supertype
}
```

Reply:
Did some changes in that direction. 
I didn't name the tests in a suite, those are compile-time checks.

https://github.com/zio/zio/pull/3241/files#diff-e70b1123afcb5d5449a81bc97a79a0d1R292-R326